### PR TITLE
Feature/fks 168

### PR DESCRIPTION
**Pull request by Evil00787**
Now displaying tasks from db in child plan details
TaskInstancesCubits
UITasks
Chips
...and much more!

### CODE REVIEW

File: lib/pages/child/plan_in_progress_page.dart [hunk 1]
```diff
@@ -18,181 +31,232 @@ class _ChildPlanInProgressPageState extends State<ChildPlanInProgressPage> {
...
-						helpPage: 'plan_info'
-					),
-					AppSegments(segments: _buildPanelSegments())
+					LoadableBlocBuilder<ChildTasksCubit>(
+						builder: (context, state) {
+							if(state is DataLoadInitial)
+								BlocProvider.of<ChildTasksCubit>(context).doLoadData();
+							else if (state is ChildTasksLoadSuccess)
+								return AppSegments(segments: _buildPanelSegments(state));
+							return Expanded(child: Center(child: CircularProgressIndicator()));
```
Comment (stasgora) [inline lib/pages/child/plan_in_progress_page.dart:44]: The point of loadableBuilder is to handle the load & progressIndicator, you only need the AppSegment case here - look at other uses

File: lib/services/data/db/mongodb_provider.dart [hunk 2]
```diff
@@ -31,7 +31,7 @@ class MongoDbProvider {
 	}
 
 	Future replaceAll(Collection collection, List<SelectorBuilder> selectors, List<Map<String, dynamic>> documents) {
-		return _execute(() => _client.collection(collection.name).updateAll(selectors, documents, upsert: true));
+		return _execute(() => _client.collection(collection.name).update(selectors, documents, upsert: true));
```
Comment (stasgora) [inline lib/services/data/db/mongodb_provider.dart:34]: Why?

File: lib/pages/child/plan_in_progress_page.dart [hunk 3]
```diff
@@ -18,181 +31,232 @@ class _ChildPlanInProgressPageState extends State<ChildPlanInProgressPage> {
...
+									getDurationChip(task),
+									getBreaksChip(task),
+									if (task.timer > 0) getTimeChip(task),
+									if (task.points.quantity != 0) getCurrencyChip(task)
+								],
+								actionButton:	ItemCardActionButton(color: Colors.teal, icon: Icons.play_arrow, onTapped: () => log("tapped start task")),
+							)
+					else if(task.taskUiType.inProgress)
+						BlocProvider<TimerCubit>(
+							create: (_) => TimerCubit(() => task.taskUiType == TaskUIType.currentlyPerformed ? sumDurations(task.duration).inSeconds : sumDurations(task.breaks).inSeconds)..startTimer(),
```
Comment (stasgora) [inline lib/pages/child/plan_in_progress_page.dart:103]: I see you created the elapsedTimePassed field but you are not using it :)

File: lib/services/task_instance_service.dart [hunk 4]
```diff
@@ -0,0 +1,59 @@
+import 'package:flutter/material.dart';
+import 'package:fokus/model/db/plan/task_instance.dart';
+import 'package:fokus/model/ui/gamification/ui_points.dart';
+import 'package:fokus/model/ui/task/ui_task_instance.dart';
+import 'package:fokus/utils/duration_utils.dart';
+import 'package:get_it/get_it.dart';
+
+import 'data/data_repository.dart';
+
+enum TaskUIType {completed, available, inBreak, stopped, currentlyPerformed, queued, notCompletedUndefined}
```
Comment (stasgora) [inline lib/services/task_instance_service.dart:10]: Move this to UITaskInstance

File: lib/model/ui/task/ui_task_instance.dart [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+	final ObjectId planInstanceId;
+	final ObjectId taskId;
+	final UIPoints points;
+	final TaskUIType taskUiType;
+	final TaskStatus status;
+	final int Function() elapsedTimePassed;
+
+  UITaskInstance(ObjectId id, String name, bool optional, String description, this.timer, this.duration, this.breaks, this.planInstanceId, this.points, this.taskId, this.taskUiType, this.status, this.elapsedTimePassed) : super(id, name, optional, description);
+	UITaskInstance.singleFromDBModel(TaskInstance task, String name, String description, UIPoints points, {int Function() elapsedTimePassed}) : this(task.id, name, task.optional, description, task.timer, task.duration, task.breaks, task.planInstanceID, points, task.taskID, TaskInstanceService.getSingleTaskInstanceStatus(task: task), task.status, elapsedTimePassed = _defElapsedTime);
+	UITaskInstance.listFromDBModel(TaskInstance task, String name, String description, UIPoints points, TaskUIType type, {int Function() elapsedTimePassed}) : this(task.id, name, task.optional, description, task.timer, task.duration, task.breaks, task.planInstanceID, points, task.taskID, type, task.status, elapsedTimePassed = _defElapsedTime);
```
Comment (stasgora) [inline lib/model/ui/task/ui_task_instance.dart:23]: When those many params are available the named parameter convention is advisable, same for the UITask

**Review by stasgora — CHANGES_REQUESTED** [review]
Round two!

File: lib/services/data/db/mongodb_provider.dart (hunk 2 above)
Comment (Evil00787) [reply lib/services/data/db/mongodb_provider.dart:34]: I don't know why it wasn't available for me, now its ok

File: lib/pages/child/plan_in_progress_page.dart (hunk 3 above)
Comment (Evil00787) [reply lib/pages/child/plan_in_progress_page.dart:103]: DAMN

File: lib/services/data/db/mongodb_provider.dart (hunk 2 above)
Comment (stasgora) [reply lib/services/data/db/mongodb_provider.dart:34]: Probably because I've just added it :)

**Review by stasgora — APPROVED** [review]

**Pull request merged**